### ISSUE TITLE
DPE Inspector: Switching Cloth components Simulate in editor toggle crashes the Editor

### ISSUE THREAD

**Issue by LB-MateuszSzweda** [o3de/o3de#16439]
**Describe the bug**
Switching the _Simulate in editor_ toggle of the **Cloth** component crashes the Editor If `ed_enableDPE` flag is enabled.

**Call stack trace**

```
25) 00007FFF4D1BDDA7 (KERNELBASE) : UnhandledExceptionFilter
24) 00007FFF4CC46A06 (InProcessClient64) : (function-name not available)
23) 00007FFF4CC46A98 (InProcessClient64) : (function-name not available)
22) 00007FFF4CC46961 (InProcessClient64) : (function-name not available)
21) 00007FFF4FA554B0 (ntdll) : memset
20) 00007FFF4FA3C886 (ntdll) : _C_specific_handler
19) 00007FFF4FA523AF (ntdll) : _chkstk
18) 00007FFF4FA014B4 (ntdll) : RtlRaiseException
17) 00007FFF4FA50EBE (ntdll) : KiUserExceptionDispatcher
16) 00007FFF277A162E (VCRUNTIME140) : memcpy
15) C:\o3de\Code\Framework\AzCore\AzCore\Serialization\Json\StringSerializer.cpp (88) : AZ::StringSerializerInternal::StoreWithDefault<AZStd::basic_string<char,AZStd::char_traits<char>,AZ::AZStdAlloc<AZ::OSAllocator> > >
14) C:\o3de\Code\Framework\AzCore\AzCore\Serialization\Json\StringSerializer.cpp (114) : AZ::JsonStringSerializer::Store
13) C:\o3de\Code\Framework\AzCore\AzCore\Serialization\Json\JsonSerializer.cpp (158) : AZ::JsonSerializer::StoreWithClassData
12) C:\o3de\Code\Framework\AzCore\AzCore\Serialization\Json\JsonSerializer.cpp (224) : AZ::JsonSerializer::StoreWithClassElement
11) C:\o3de\Code\Framework\AzCore\AzCore\Serialization\Json\JsonSerializer.cpp (261) : AZ::JsonSerializer::StoreClass
10) C:\o3de\Code\Framework\AzCore\AzCore\Serialization\Json\JsonSerializer.cpp (156) : AZ::JsonSerializer::StoreWithClassData
 9) C:\o3de\Code\Framework\AzCore\AzCore\Serialization\Json\JsonSerializer.cpp (224) : AZ::JsonSerializer::StoreWithClassElement
 8) C:\o3de\Code\Framework\AzCore\AzCore\Serialization\Json\JsonSerializer.cpp (261) : AZ::JsonSerializer::StoreClass
 7) C:\o3de\Code\Framework\AzCore\AzCore\Serialization\Json\JsonSerializer.cpp (156) : AZ::JsonSerializer::StoreWithClassData
 6) C:\o3de\Code\Framework\AzCore\AzCore\Serialization\Json\JsonSerializer.cpp (70) : AZ::JsonSerializer::Store
 5) C:\o3de\Code\Framework\AzCore\AzCore\Serialization\Json\JsonSerialization.cpp (322) : AZ::JsonSerialization::Store
 4) C:\o3de\Code\Framework\AzCore\AzCore\std\function\function_template.h (172) : AZStd::Internal::function_util::get_invoker<AZ::Dom::Value __cdecl(AZ::Dom::Value const &),AZStd::Internal::function_util::function_obj_tag,AZStd::allocator>::call<`AZ::Documen
 3) C:\o3de\Code\Framework\AzCore\AzCore\std\function\function_template.h (172) : AZStd::Internal::function_util::get_invoker<void __cdecl(AZ::Dom::Value const &,enum AZ::DocumentPropertyEditor::Nodes::ValueChangeType),AZStd::Internal::function_util::functio
 2) C:\o3de\Code\Framework\AzFramework\AzFramework\DocumentPropertyEditor\DocumentAdapter.h (49) : AZ::DocumentPropertyEditor::AdapterMessage::Match<AZ::DocumentPropertyEditor::CallbackAttributeDefinition<void __cdecl(AZ::Dom::Value const &,enum AZ::Document
 1) C:\o3de\Code\Framework\AzFramework\AzFramework\DocumentPropertyEditor\ReflectionAdapter.cpp (1181) : AZ::DocumentPropertyEditor::ReflectionAdapter::HandleMessage
```

Please note that only switching the toggle on or off results in a crash. The cloth behavior can still be simulated by using the Game or Simulate mode.

The crash will occur regardless if any node derived from Actor or Mesh component is selected in the Cloth component.

The Issue occurs on both Windows and Linux platforms.

For further information please refer to the attached logs and video file.

**Steps to reproduce**

1. Launch the Editor and load any level.
2. Enable DPE in Editor by inputting _ed_enableDPE=true_ command in the Console.
3. Create a new entity and add **Cloth** and **Mesh** components to it.
4. Switch on the _Simulate in editor_ toggle in the **Cloth** component.

**Expected behavior**
Editor remains stable.

**Actual behavior**
Editor crashes upon switching the _Simulate in editor_ toggle.

**Screenshots/Video**

https://github.com/o3de/o3de/assets/86953456/3fac8baf-2b6a-4a04-a748-bf09cd73c9c5

**Found in Branch**
Development

**Commit ID from** [o3de/o3de](https://github.com/o3de/o3de) **Repository**
3210473

**Desktop/Device:**
- Device: PC
- OS: Windows
- Version: 10 Pro
- CPU: AMD Ryzen 9 3900X
- GPU: NVIDIA GeForce RTX 2060 SUPER
- Memory: 16GB

**Additional context**
[error.zip](https://github.com/o3de/o3de/files/12160627/error.zip)
[error.log](https://github.com/o3de/o3de/files/12160630/error.log)
[Editor.log](https://github.com/o3de/o3de/files/12160631/Editor.log)

**Comment by alexmontAmazon** [comment]
JSon related...


![Image](https://github.com/o3de/o3de/assets/85521892/e1e535d0-6ece-4028-8810-00be3cb4ef89)

**Comment by LB-MateuszSzweda** [comment]
Issue verified as fixed in Stabilization/2310 (21bc690) and Linux stabilization installer. However it is worth noting that the issue still occured on Windows Stabilization/2310 installer build downloaded at around 8AM CEST/11PM PST.
